src/Microsoft.Extensions.Configuration.FileProviderExtensions/ConfigurationRootExtensions.cs

```diff
@@ -20,8 +21,13 @@ public static IConfigurationRoot ReloadOnChanged(this IConfigurationRoot config,
                 throw new ArgumentNullException(nameof(filename));
             }
 #if NET451
-            var basePath = AppDomain.CurrentDomain.GetData("APP_CONTEXT_BASE_DIRECTORY") as string ??
-                AppDomain.CurrentDomain.BaseDirectory ?? string.Empty;
+            var basePath =
```

Comment:
> Currently we require users to provide the location and name of the file again to setup auto-reloading of the configuration if the file changes.

I think it would be nice if we could make setting up the reload part of adding the files. We wanted to massage the startup code more when we built it, but due to the lateness of it in rc1 we didn't want to impact existing APIs so it was purely additive.

I'll try to come up with something more all-encompasing that can eliminate the conceptual duplication. That's really outside of the scope of this change though.